react-app/src/components/InputText/InputText.js

```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react';
+import FontAwesomeIcon from '@fortawesome/react-fontawesome';
+import { faEye, faEyeSlash } from '@fortawesome/fontawesome-free-solid';
+
+import './InputText.css';
+
+export default class InputText extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      type: this.props.type,
+      togglePasswordVisibilityButtonIcon: faEye
+    };
+
+    this.togglePasswordVisibility = this.togglePasswordVisibility.bind(this);
+  }
+
+  togglePasswordVisibility() {
+    if (this.state.type === 'password') {
+      this.setState({
+        type: 'text',
+        togglePasswordVisibilityButtonIcon: faEyeSlash
+      });
+    } else {
+      this.setState({
+        type: 'password',
+        togglePasswordVisibilityButtonIcon: faEye
+      });
+    }
+  }
+
+  render() {
+    const togglePasswordVisibilityButton = (this.props.type === 'password') ? (
```

Comment:
Have in mind that for some cases (this isn't necessarily one), it makes sense to extract part of the UI generation into a function so the `render` method is simpler and has less variables "lying around". If this were the case we could have a `const icon = this._getIcon();` function, for instance.